Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Customers.m

```diff
@@ -0,0 +1,269 @@
+}
+
+- (NSURLSessionDataTask *)recoverPasswordForCustomer:(NSString *)email callback:(BUYDataCheckoutStatusBlock)block
+{
+	NSURLComponents *components = [self URLComponentsForPasswordReset];
+	
+	return [self postRequestForURL:components.URL object:@{@"email": email} completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
+		
+		NSInteger statusCode = [(NSHTTPURLResponse *)response statusCode];
+		error = error ?: [self extractErrorFromResponse:response json:json];
+		
+		block(statusCode, error);
+	}];
+}
+
+- (NSURLSessionDataTask *)renewCustomerTokenWithID:(NSString *)customerID callback:(BUYDataTokenBlock)block
+{
+	if (self.customerToken) {
+		NSURLComponents *components = [self URLComponentsForTokenRenewalWithID:customerID];
+		
+		return [self putRequestForURL:components.URL body:nil completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
+			
+			NSString *accessToken = nil;
+			if (json && error == nil) {
+				BUYAuthenticatedResponse *authenticatedResponse = [BUYAuthenticatedResponse responseFromJSON:json];
+				accessToken = authenticatedResponse.accessToken;
+			}
+			
+			error = error ?: [self extractErrorFromResponse:response json:json];
+			
+			block(accessToken, error);
+		}];
+	}
+	else {
+		block(nil, [NSError errorWithDomain:kShopifyError code:BUYShopifyError_InvalidCustomerToken userInfo:nil]);
+		return nil;
+	}
+}
+
+- (NSURLSessionDataTask *)activateCustomerWithCredentials:(BUYAccountCredentials *)credentials customerID:(NSString *)customerID customerToken:(NSString *)customerToken callback:(BUYDataCustomerTokenBlock)block
+{
+	NSURLComponents *components = [self URLComponentsForCustomerActivationWithID:customerID customerToken:customerToken];
+	NSData *data = [NSJSONSerialization dataWithJSONObject:credentials.JSONRepresentation options:0 error:nil];
+	
+	return [self putRequestForURL:components.URL body:data completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
+		if (json && error == nil) {
+			BUYAccountCredentialItem *emailItem = [BUYAccountCredentialItem itemWithKey:@"email" value:json[@"customer"][@"email"]];
+			credentials[@"email"] = emailItem;
+			[self loginCustomerWithCredentials:credentials callback:block];
+		}
+		else {
+			block(nil, nil, error);
+		}
+	}];
+}
+
+- (NSURLSessionDataTask *)resetPasswordWithCredentials:(BUYAccountCredentials *)credentials customerID:(NSString *)customerID customerToken:(NSString *)customerToken callback:(BUYDataCustomerTokenBlock)block
+{
+	NSURLComponents *components = [self URLComponentsForCustomerPasswordResetWithCustomerID:customerID customerToken:customerToken];
+	NSData *data = [NSJSONSerialization dataWithJSONObject:credentials.JSONRepresentation options:0 error:nil];
+	
+	return [self putRequestForURL:components.URL body:data completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
+		if (json && error == nil) {
```

Comment:
Do you mind going through these completion blocks and updating them to be consistent in how we check for an error and/or JSON? Sometimes we check against `nil`, sometimes not, sometimes we check the error first, sometimes the JSON.